tests/test_views.py

```diff
@@ -133,4 +155,20 @@ def test_full_upload_flow(api_client: APIClient, file_size: int):
         'field_value': Re(r'.*:.*'),
     }
 
+    # Verify that the Content headers were stored correctly on the object
+    object_resp = requests.get(default_storage.url(initialization['object_key']))
+    assert resp.status_code == 200
+    actual_content_type = object_resp.headers['Content-Type']
+    actual_content_disposition = object_resp.headers['Content-Disposition']
+    if content_type is not None:
+        assert actual_content_type == content_type
+    else:
+        # Test Content-Type inference when none is specified
+        assert actual_content_type == 'text/plain'
+    if content_disposition is not None:
+        # Test default Content-Disposition when none is specified
+        assert actual_content_disposition == content_disposition
+    else:
+        assert actual_content_disposition == 'attachment; filename="test.txt"'
```

Comment:
This test case is getting very complex, which could make it expensive to maintain. Is there a sane way to break it up a bit?

( @brianhelba may have something to say about this )

Reply:
Copying the test for each use case would just double the complexity, so I've been avoiding it.

The only solution I can think of would be a test helper that does most of the upload process on it's own, which I consider something of an antipattern. I'm open to suggestions though, this is a very bloated test.

Reply:
The body itself is big and branchy, but also just the cardinality of the cartesian product of all these `parametrize` decorators is getting big. That's fine if we really want to test that whole space, but if not, it'd be good to identify the most important cases and maybe collapse the decorators some.

Reply:
I agree, I think we should rework this whole test. I'm ok with merging the PR with this, as long as we refactor things at some point.

Reply:
I'm not 100% sure but it looks like maybe @dchiquito already went through and removed some of the complexity here. It looks less scary to me now, at least in terms of this changeset.